src/c++/perf_analyzer/genai-perf/genai_perf/parser.py

```diff
@@ -147,25 +161,42 @@ def _check_conditional_args(
                 "The --output-tokens-mean-deterministic option is only supported with the Triton service-kind."
             )
 
-    _check_conditional_args_embeddings(parser, args)
+    _check_conditional_args_embeddings_rankings(parser, args)
 
     return args
 
 
-def _check_conditional_args_embeddings(
+def _check_conditional_args_embeddings_rankings(
     parser: argparse.ArgumentParser, args: argparse.Namespace
 ):
-    if args.endpoint_type == "embeddings":
+
+    if args.output_format in [
+        OutputFormat.OPENAI_EMBEDDINGS,
+        OutputFormat.RANKINGS,
+    ]:
         if args.streaming:
             parser.error(
-                "The --streaming option is not supported with the embeddings endpoint type."
+                "The --streaming option is not supported with the embeddings nor rankings endpoint type."
```

Comment:
```suggestion
                f"The --streaming option is not supported with the {args.endpoint_type} endpoint type."
```

Reply:
Removing parser.error suggestion here, since this function doesn't have access to the parser. We could clean it up in a future ticket.